.github/workflows/tests.yml

```diff
@@ -0,0 +1,35 @@
+---
+# Runs Tests on Pushes to `master` and Pull Requests
+name: Tests
+on:
+  push:
+    branches:
+      - master
+    paths:
+      - '**.c'
+      - '**.h'
+  pull_request:
+    paths:
+      - '**.c'
+      - '**.h'
+jobs:
+  unit-tests:
+    name: Unit Tests
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          # We need a full deep clone for the checks done by the unit testing script to work.
```

Comment:
I don't see why we need a deep clone for running unit tests. Perhaps you wanted to say that we _don't_ need a full deep clone for those checks.

Reply:
This is a migration from the old Travis workflows. Its possible this comment is invalid/obsolte? I'm not sure at the moment... I didn't look into it too much yet.

Reply:
@cosmix We need the deep clone so we can skip them if nothing that we actually run unit tests for has changed.

The unit tests take a _very_ long time to run because they include stress testing for the dbengine (more than 20 minutes total most of the time), and end up delaying PR check completion for quite some time as a result. We should only be running them if things being tested are changed, and ideally should deaggregate them so that we can test only changes.

Reply:
> @cosmix We need the deep clone so we can skip them if nothing that we actually run unit tests for has changed.

No I removed all that code and am using the GHA path filters themselves. Much eaiser.

Reply:
We still don't know why we need a full depth clone though

Reply:
So does it not work if we don't do a full-depth clone then?

Reply:
Don't know but let's find out :)